nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/DeleteMongo.java

```diff
@@ -0,0 +1,151 @@
+            .displayName("Delete Mode")
+            .description("Choose between deleting one document by query or many documents by query.")
+            .allowableValues(DELETE_ONE, DELETE_MANY)
+            .defaultValue("one")
+            .addValidator(Validator.VALID)
+            .build();
+
+    static final PropertyDescriptor FAIL_ON_NO_DELETE = new PropertyDescriptor.Builder()
+            .name("delete-mongo-fail-on-no-delete")
+            .displayName("Fail When Nothing Is Deleted")
+            .description("Determines whether to send the flowfile to the success or failure relationship if nothing is successfully deleted.")
+            .allowableValues(YES_FAIL, NO_FAIL)
+            .defaultValue("true")
+            .addValidator(Validator.VALID)
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("All FlowFiles that are written to MongoDB are routed to this relationship").build();
+    static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("All FlowFiles that cannot be written to MongoDB are routed to this relationship").build();
+
+    static {
+        List<PropertyDescriptor> _propertyDescriptors = new ArrayList<>();
+        _propertyDescriptors.addAll(descriptors);
+        _propertyDescriptors.add(DELETE_MODE);
+        _propertyDescriptors.add(FAIL_ON_NO_DELETE);
+        _propertyDescriptors.add(WRITE_CONCERN);
+        propertyDescriptors = Collections.unmodifiableList(_propertyDescriptors);
+
+        final Set<Relationship> _relationships = new HashSet<>();
+        _relationships.add(REL_SUCCESS);
+        _relationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(_relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final WriteConcern writeConcern = getWriteConcern(context);
+        final MongoCollection<Document> collection = getCollection(context).withWriteConcern(writeConcern);
+        final String deleteMode = context.getProperty(DELETE_MODE).getValue();
+        final Boolean failMode  = context.getProperty(FAIL_ON_NO_DELETE).asBoolean();
+
+        DeleteResult result;
+
+        try {
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            session.exportTo(flowFile, bos);
+            bos.close();
+
+            String json = new String(bos.toByteArray());
+            Document query = Document.parse(json);
+
+            getLogger().info("Test\n\n\n\n\n\n");
```

Comment:
This can be removed (or replaced with a debug/info message to inform what's going on